src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -969,6 +986,56 @@ CHIP_ERROR BLEManagerImpl::InitESPBleLayer(void)
     return err;
 }
 
+void BLEManagerImpl::DeinitESPBleLayer()
+{
+    VerifyOrReturn(DeinitBLE() == CHIP_NO_ERROR);
+    BLEManagerImpl::ClaimBLEMemory(nullptr, nullptr);
+}
+
+void BLEManagerImpl::ClaimBLEMemory(System::Layer *, void *)
+{
+    TaskHandle_t handle = xTaskGetHandle("nimble_host");
+    if (handle)
+    {
+        ChipLogDetail(DeviceLayer, "Schedule ble memory reclaiming since nimble host is still running");
+
+        // Rescheduling it for later, 2 seconds is an arbitrary value, keeping it a bit more so that
+        // we dont have to reschedule it again
+        SystemLayer().StartTimer(System::Clock::Seconds32(2), ClaimBLEMemory, nullptr);
```

Comment:
The first execution of the function is through the caller of `DeinitESPBleLayer`, right? Can it be any task?

Also, the subsequent executions are through the timer task?

Reply:
The trigger for this is `_Shutdown()`, it sets some flags which tells the state machine to deinit and reclaim memory. Every action from there will always run in the chip thread context.